sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -309,13 +273,13 @@ std::unique_ptr<RawResponse> CurlTransport::Send(Request& request, Context const
             request,
             m_options,
             getConnectionOpenIntent + 1 >= _detail::RequestPoolResetAfterConnectionFailed),
-        m_options.HttpKeepAlive);
+        m_options);
   }
 
   if (performing != CURLE_OK)
   {
-    throw Azure::Core::Http::TransportException(
-        "Error while sending request. " + std::string(curl_easy_strerror(performing)));
+    throw Azure::Core::Http::CurlTransportException(
```

Comment:
this bubbles back to the pipeline, who doesn't know about what's the transport.
then it bubbles to the sdk client, who knows about the transport but should not be catching exceptions from specific transports.
Then it bubbles to customer, who can know be confused about when to catch the TransportException and when to catch the specific transport adapter exception.

Reply:
If the customer cares about catching but doesn't know about the transport, then they catch `TransportException`, which works just fine (since `CurlTransportException` derives from `TransportException`). 

The only customer scenarios where this type comes into play is when the customer explicitly knows they're dealing with a CURL or WinHTTP transport.